mllib-local/src/main/scala/org/apache/spark/ml/linalg/BLAS.scala

```diff
@@ -27,15 +27,35 @@ private[spark] object BLAS extends Serializable {
 
   @transient private var _f2jBLAS: NetlibBLAS = _
   @transient private var _nativeBLAS: NetlibBLAS = _
+  @transient private val vectorSizeThreshold: Int = 256
```

Comment:
This does not need to be transient. It's an int.

Reply:
`256` depends on runtime environment such as CPU spec, JVM version, and so on.   
It should be taken from the property.

Reply:
@srowen I think @kiszk make a good point. With properly tuned based on specific workload and runtime environment, nativeBLAS may can bring much benefit than f2jBLAS even vector size < 256.  Making it configurable can provide user a simpler way to use nativeBLAS in Spark ML when vector size is not so big. Do you think so? if so, we need to add spark-core dependency in spark-mllib-local package.
Thanks @kiszk for your comments.